src/wallet/api.rs

```diff
@@ -1255,63 +1260,140 @@ impl Wallet {
+            return Ok(single_group.clone());
+        }
+
+        // Multi-group selection: accumulate groups until target+fees is met
+        let mut pre_selected_utxos = Vec::new();
+        let mut pre_selected_total = 0u64;
+        let mut all_groups = single_addresses;
+
+        for group in grouped_addresses {
+            let group_total: u64 = group
+                .iter()
+                .map(|(u, _): &(ListUnspentResultEntry, UTXOSpendInfo)| u.amount.to_sat())
+                .sum();
+
+            // Check if we need more funds
+            let current_estimated_fee = if pre_selected_utxos.is_empty() {
+                0
+            } else {
+                estimate_group_fee(&pre_selected_utxos).unwrap_or(0)
+            };
+
+            if pre_selected_total < target_amount + current_estimated_fee {
+                log::info!(
+                    "Address grouping: Pre-selecting {} UTXOs (total: {} sats)",
+                    group.len(),
+                    group_total
+                );
+                pre_selected_total += group_total;
+                pre_selected_utxos.extend(group.iter().cloned());
             }
+            all_groups.push(group);
         }
 
-        let grouped_utxos = address_groups
-            .into_iter()
-            .map(|(_, utxos)| utxos)
-            .collect::<Vec<_>>();
+        // Check if pre-selected groups cover target+fees
+        let final_estimated_fee = estimate_group_fee(&pre_selected_utxos).unwrap_or(0);
+        if pre_selected_total >= target_amount + final_estimated_fee {
+            log::info!("Address grouping covers target+fees - returning {} pre-selected UTXOs (total: {} sats, target+fee: {} sats)", 
+                pre_selected_utxos.len(), pre_selected_total, target_amount + final_estimated_fee);
+            return Ok(pre_selected_utxos);
+        }
 
-        let output_groups = grouped_utxos
+        // Run coin selection on all groups
+        let output_groups = all_groups
             .iter()
-            .map(|utxos_in_group| {
-                let total_value: u64 = utxos_in_group
+            .map(|group| {
+                let total_value: u64 = group
                     .iter()
-                    .map(|(utxo, _)| utxo.amount.to_sat())
+                    .map(|(utxo, _): &(ListUnspentResultEntry, UTXOSpendInfo)| utxo.amount.to_sat())
                     .sum();
-                let total_weight: u64 = utxos_in_group
+                let total_weight: u64 = group
                     .iter()
                     .map(|(_, spend_info)| 36 + spend_info.estimate_witness_size() as u64)
                     .sum();
 
                 OutputGroup {
                     value: total_value,
                     weight: total_weight,
-                    input_count: utxos_in_group.len(),
+                    input_count: group.len(),
                     creation_sequence: None,
                 }
             })
```

Comment:
This is not correct anymore. 

We should only run CS on single addresses. We have already selected the grouped utxos by this point. Sending them again in CS will be a bug, CS main doesn't understand groupings anyway.

In a way the `OutputGroup` structure is redundant now. We are handling group selection at pre-processing, and passing groups to CS has no effect. Currently this can cause double selection of the same grouped utxos.

Later we should simplify the CS API to either remove this grouping schenanigan and just give a list of single utxos, or add this grouping logic inside the CS main.

For now the CS should only work on single-address utxos.